lib/URI/oracle.pm

```diff
@@ -5,13 +5,28 @@ our $VERSION = '0.22';
 sub default_port { 1521 }
 sub dbi_driver   { 'Oracle' }
 
-sub _dbi_param_map {
+sub _dsn_params {
     my $self = shift;
-    return (
-        [ host => scalar $self->host   ],
-        [ port => scalar $self->_port  ],
-        [ service_name => scalar $self->dbname ],
-    );
+    my $name = $self->dbname || '';
+    my $dsn = $self->host;
+
+    if (my $p = $self->_port) {
+        $dsn .= ":$p";
+    }
+
+    return $name unless $dsn;
+    $dsn .= "/$name";
+
+
+    if (my @p = $self->query_params) {
+        my @kvpairs;
+        while (@p) {
+            push @kvpairs => join '=', shift @p, shift @p;
+        }
+        $dsn .= '?' . join ';' => @kvpairs;
```

Comment:
Looking at the EZCONNECT syntax, it looks like parameters should be separated by `&`. Don't ask me what to do if a parameter or value contains `&`.

Reply:
Fixed in bfbd9f8, thanks.